datascience/table.py

```diff
@@ -543,9 +551,9 @@ def index_by(self, column_or_label):
         rows corresponding to each value.
         """
         column = self._get_column(column_or_label)
-        index = {}
+        index = collections.defaultdict(list)
         for key, row in zip(column, self.rows):
-            index.setdefault(key, []).append(row)
+            index[key].append(row)
         return index
```

Comment:
`return dict(index)` so that we don't leak `defaultdict`s out into the world (or just revert).